### ISSUE TITLE
No operations defined in spec! while loading the JSON file

### ISSUE THREAD

**Issue by NagenKumarSahu** [swagger-api/swagger-ui#2757]
Hi,
I am able to create the swagger.json file. But when i want to show it in the UI, I am facing the "No operations defined in spec!" issue. 

I have done the following changes in the public/index.html file

window.onload = function() {
  // Build a system
  const ui = SwaggerUIBundle({
    //url: "http://petstore.swagger.io/v2/swagger.json",
    url: "http://localhost:9080/E2EVisibility/swagger.json",
    dom_id: '#swagger-ui',
    presets: [
      SwaggerUIBundle.presets.apis,
      // yay ES6 modules ↘
      Array.isArray(SwaggerUIStandalonePreset) ? SwaggerUIStandalonePreset : SwaggerUIStandalonePreset.default
    ],
    plugins: [
      SwaggerUIBundle.plugins.DownloadUrl
    ],
    layout: "StandaloneLayout",
    validatorUrl: "false"
  })

  window.ui = ui
}

Please help I am new to swagger implementation.

**Comment by NagenKumarSahu** [comment]
While loading the JSON it is showing the following error in the browser console

![swaggererror](https://cloud.githubusercontent.com/assets/16772047/24145123/31d4ca78-0e06-11e7-8036-2bf4bbbd3090.png)

**Comment by webron** [comment]
If you open http://localhost:9080/E2EVisibility/swagger.json in your browser, is it accessible?

**Comment by beriba** [comment]
I have the same issue. YML doc that I created is definitely visible, online validation shows the error when I make it intentionally. I don't have any errors in the console. I'm using swagger-ui v3.0.2. The same doc works on swagger-ui v2.1.4. I have CORS enabled and it also works because the same thing happens when using doc on petstore.swagger.io. Here you have the minimum doc to reproduce this. I tried to compare it to default petstore doc but I can't see anything that could cause the problem.

`swagger: '2.0'                                                                                                          
info:                                                                                                                   
  description: API doc                                                                                                  
  version: 1.0.0                                                                                                        
  title: The API                                                                                                        
  termsOfService: 'blah'                                                                                                
  contact:                                                                                                              
    email: admin@example.com                                                                                            
  license:                                                                                                              
    name: 'Private'                                                                                                        
    url: 'http://example.com'                                                                                           
host: api.example.com                                                                                                   
basePath: /                                                                                                             
tags:                                                                                                                      
    - name: Search                                                                                                      
      description: dasdasd                                                                                              
schemes:                                                                                                                
  - http                                                                                                                
paths:                                                                                                                     
  /search:                                                                                                              
    get:                                                                                                                   
      tags:                                                                                                                
        - Search                                                                                                           
      operationId: search                                                                                                  
      summary: Search                                                                                                      
      description: Search                                                                                                  
      consumes:                                                                                                            
        - application/json                                                                                                 
      produces:                                                                                                            
        - application/json                                                                                                 
      parameters:                                                                                                          
        - name: query                                                                                                      
          in: query                                                                                                        
          required: true                                                                                                   
          type: string                                                                                                     
          description: Query                                                                                               
      responses:                                                                                                           
        200:                                                                                                               
          description: Found results`

Am I doing something wrong or is it a bug in swagger-ui?

**Comment by NagenKumarSahu** [comment]
@webron  I am able to access the swagger.json file while clicking it in the url.

the strange thing is , everything is working fine in the google chrome except the error. But it is not working in the firefox browser.

firefox version: 45.8.0

**Comment by webron** [comment]
@NagenKumarSahu you're using an old version of Firefox. Try upgrading it and check again.

**Comment by NagenKumarSahu** [comment]
Thanks @webron for the information.

**Comment by valyke** [comment]
not working to me. im using chrome

**Comment by valyke** [comment]
im getting the same message. How did u fixed it?

**Comment by beriba** [comment]
I asked new question here #2824

**Comment by nosisky** [comment]
I had this same issue but was able to fix it by fixing the typo in my json file.

**Comment by FabianPiconeDev** [comment]
I had the same issue, it was a typo in `path="/MyProject/MyClass"` in `@SWG\Get(...)`.

**Comment by Dipthi123** [comment]
I am getting as well same issue No operations defined in spec!. I am trying to setup swagger UI only for documentation. I have downloaded latest swagger UI from git. I have json file given by client.  I kept in one folder the json file, 
started http-server --cors. Now my json file is available via url http://localhost:8080/MyJson.json, now i gave this to index html and click on explore. It gives No operations defined in spec.
[Json file indivisually accessed http://localhost:8080/MyJson.json gives no errors and shown well]
Is there some step I am missing ? I am new to swagger UI, your help is really appreciated.
